src/authorization/WebAclAuthorizer.ts

```diff
@@ -81,28 +87,88 @@ export class WebAclAuthorizer extends Authorizer {
    * @param agent - Agent whose credentials will be used for the `user` field.
    * @param acl - Store containing all relevant authorization triples.
    */
-  private createAuthorization(agent: Credentials, acl: Store): WebAclAuthorization {
-    const publicPermissions = this.determinePermissions({}, acl);
-    const userPermissions = this.determinePermissions(agent, acl);
+  private async createAuthorization(agent: Credentials, acl: Store): Promise<WebAclAuthorization> {
+    // Get public permissions
+    const publicPermissions = this.determinePublicPermissions(acl);
```

Comment:
After having gone through the code I don't think splitting `determinePermissions` up into two different functions is the way to go. `determinePublicPermissions` now implements some of the access check behaviour while `determineAgentPermissions` has offloaded that behaviour to the access checkers.

My suggestion would be to revert back to a single `determinePermissions` function that always uses the access checkers. I'm guessing the main reason for this split is because you don't want any heavy fetch calls to happen for the public part, but if you check the presence of the webid in those handlers then this is not going to be an issue, and each checker supports all agents, not just those with an identified WebID.

We would also need a checker to check the public agent class. I suggest renaming `AuthenticatedAccessChecker` to `AgentClassAcessChecker`, always checking for `FOAF.Agent` and only checking for `ACL.AuthenticatedAgent` if the `webid` field is defined.